fixes/consolidate-multiple-customtextfield-history.sql

```diff
@@ -0,0 +1,124 @@
+from H as aH
+join ' + @customFieldTable + ' cField on cField.ID=aH.ID and aH.AuditID = cField.AuditBegin
+join H as bH on bH.ID=aH.ID and bH.R=aH.R+1
+join ' + @customFieldTable + ' bCF on bCF.ID=bH.ID and bH.AuditID = bCF.AuditBegin'
+
+-- Consecutive Custom Field value changes
+declare @q2 varchar(max);
+select @q2 = ';
+insert #suspect(ID, CurrentAuditID, NextAuditID)
+select Assets.ID, Assets.CurrentAuditID, Assets.NextAuditID
+from #customfield as Assets
+join ' + @customFieldTable + ' currentCF on currentCF.ID=Assets.ID and currentCF.AuditBegin=Assets.CurrentAuditID
+join ' + @customFieldTable + ' nextCF on nextCF.ID=Assets.ID and nextCF.AuditBegin=Assets.NextAuditID
+join dbo.[Audit] currentA on currentA.ID = Assets.CurrentAuditID
+join dbo.[Audit] nextA on nextA.ID = Assets.NextAuditID
+WHERE
+currentCF.[Definition] = ''' + @fieldName + ''' AND nextCF.[Definition] = ''' + @fieldName + ''' -- Targetted field type
+AND ISNULL(currentA.[ChangedByID] ,-1) = ISNULL(nextA.[ChangedByID],-1)  -- Consecutive changes from the same user
+AND DATEDIFF(mi,currentA.[ChangeDateUTC] ,nextA.[ChangeDateUTC]) <= ' + CAST(@timethreshold as varchar(10)) + ' --Time threshold / period / lapse.
+AND	ISNULL(currentCF.[Value],-1) != ISNULL(nextCF.[Value],-1) --Value has changed'
+
+-- consecutive redundant CustomText values
+declare @q3 varchar(max);
+select @q3 = ';
+insert #bad(ID, CurrentAuditID, NextAuditID)
+select _.ID, CurrentAuditID, NextAuditID
+from #suspect _
+join ' + @customFieldTable + ' A on A.ID=_.ID and A.AuditBegin=_.CurrentAuditID
+join ' + @customFieldTable + ' B on B.ID=_.ID and B.AuditBegin=_.NextAuditID
+join ' + @customFieldTable + ' C on C.ID=_.ID and C.AuditEnd=_.CurrentAuditID AND ISNULL(C.[Value],-1) != ISNULL(A.[Value],-1)'
+
+-- rows to purge
+declare @q4 varchar(max);
+select @q4 = ';
+select cf.ID, cf.AuditBegin, a.ChangedByID, a.ChangeDateUTC
+from #bad b
+join ' + @customFieldTable + ' cf ON cf.ID=b.ID and cf.AuditBegin=b.CurrentAuditID
+join dbo.[Audit] a on a.ID = b.CurrentAuditID'
+
+-- purge redundant rows
+declare @q5 varchar(max);
+select @q5 = ';
+delete ' + @customFieldTable +
+' from #bad
+where ' + @customFieldTable + '.ID=#bad.ID and (' + @customFieldTable + '.AuditBegin=#bad.CurrentAuditID)'
+
+declare @error int, @rowcount int
+
+select @rowcount=@@ROWCOUNT, @error=@@ERROR
+if @error<>0 goto ERR
+raiserror('%d CustomField value historical records consolidated', 0, 1, @rowcount) with nowait
+
+if @rowcount=0 goto FINISHED
+
+-- re-stitch CustomField history
+declare @q6 varchar(max)
+select @q6 = ';
+with H as (
+	select ID, AuditBegin, AuditEnd, R=ROW_NUMBER() over(partition by ID order by AuditBegin)
+	from ' + @customFieldTable +
+')
+update ' + @customFieldTable + ' set AuditEnd=B.AuditBegin
+from H A
+left join H B on A.ID=B.ID and A.R+1=B.R
+where ' + @customFieldTable + '.ID=A.ID and ' + @customFieldTable + '.AuditBegin=A.AuditBegin
+	and isnull(A.AuditEnd,-1)<>isnull(B.AuditBegin,-1)'
```

Comment:
This is not going to work, for the same reason as above - custom tables history is not contiguous - i.e. `prior.AuditEnd=next.AuditBegin` does not hold true.